src/store/slices/profile.slice.ts

```diff
@@ -19,35 +20,83 @@ export interface UserProfile {
     subject?: string;
     githubToken?: string;
   } | null;
+  role: string;
 }
 
 interface ProfileState {
   profile: UserProfile | null;
   errorMessage: string | null;
   loading: boolean;
   success: boolean;
+  selectedUser: any;
+  userSubscription: any;
+  allUsers: any;
+  loadingDetails: boolean;
+  loadingSubDetails: boolean;
+  detailsSuccess: boolean;
+  updateSuccess: boolean;
+  subDetailsSuccess: boolean;
+  detailsError: string | null;
+  subDetailsError: string | null;
+  loadingHistory: boolean;
+  runHistory: Run[];
+  impersonate: boolean;
+  retainId: any;
+}
+
+export interface IUpdateProfile {
+  companyName: string;
+  contactEmail: string;
+  dapp: {
+    githubToken: string;
+    name: string;
+    owner: string;
+    repo: string;
+    subject: string;
+    version: string;
+  } | null;
+  email: string;
+  fullName: string;
+  id: number;
+  linkedin: string;
+  twitter: string;
+  website: string;
 }
 
 const initialState: ProfileState = {
   profile: null,
   errorMessage: null,
   loading: false,
   success: false,
+  selectedUser: null,
+  allUsers: null,
+  loadingDetails: false,
+  detailsSuccess: false,
+  detailsError: null,
+  userSubscription: null,
+  loadingSubDetails: false,
+  subDetailsSuccess: false,
+  subDetailsError: null,
+  updateSuccess: false,
+  loadingHistory: false,
+  runHistory: [],
+  impersonate: false,
+  retainId: null
 };
 
 export const fetchProfile = createAsyncThunk('fetchProfile', async (payload: {}, thunkApi) => {
   try {
-    const response = await fetch<UserProfile>(thunkApi, { method: 'GET', url: '/profile/current' });
+    const response = await fetch<UserProfile>(thunkApi, { method: 'GET', url: `/profile/${payload || 'current'}` });
```

Comment:
We can apply the same principle like the one I suggested for the `createTestRun` action here
Mostly because the `profileId` can be obtained from a redux store state instead of asking it as a function parameter
But in case that you want to override the `profileId` from the state you can add it as an optional parameter inside the payload, like this `payload: { profileId?: string }`